### ISSUE TITLE
Deployment stuck without error message

### ISSUE THREAD

**Issue by SomeBdyElse** [sebafo/frontdoor-container-apps#2]
When I try to deploy to Azure Cloud, the deployment gets stuck forever, without any error message. 

I can see the rg, log, vnet and container app environment beeing created, but the actual app is missing. After hours, the deployment terminates with a timeout. 

Azure support advised that it might have to do with a missing network security rule, to allow the communication between ips of the subnet. Yet, I was not able to verify and so for they were not able to fix it.

**Comment by sebafo** [comment]
Thanks for your feedback.

Which Azure region have you configured? Does it work if you deploy Container Apps in the Azure portal in this region? Have you tried a different region?
Have you updated az-cli and bicep to the most recent version (“az bicep upgrade”)?

As a quick test I have started the script and deployed it successfully. The deployment lasted around 16 minutes in westeurope.
<img width="1507" alt="Bildschirm­foto 2022-12-22 um 14 46 20" src="https://user-images.githubusercontent.com/72460768/209149184-7326af58-417f-4239-9cdd-e912d2de0bdd.png">



You can access this information in your resource group in the “Settings” section in the “Deployments” blade.

If it sticks at the Container Apps environment step (containerapps), could you perhaps verify if there is a resource group named like “MC_something-identifier-rg_something-identifier_region”. This rg is created by the Container Apps environment and should contain 4 visible resources:
- Public IP address
- “kubernetes”-Loadbalancer (both for outgoing traffic)
- Network security group for the aks-agentpool
- “kuberentes-internal” loadbalancer (ingress, that is connected via private link to Azure Frontdoor)

<img width="754" alt="Bildschirm­foto 2022-12-22 um 15 03 52" src="https://user-images.githubusercontent.com/72460768/209151041-2e679999-9c9b-4ce9-914b-87ef2414ea88.png">

**Comment by SomeBdyElse** [comment]
OK, good to know that the deployment works for you. This indicates there might be something wrong on my end (subscription, policies, region, …). 

I just checked the deployment status. 
<img width="1479" alt="Screenshot 2022-12-22 at 15 09 34" src="https://user-images.githubusercontent.com/243504/209152217-bbf4fbcc-fdbd-44da-bd50-9a0eb154ea50.png">

Actually, I do not see a resource group named like "MC_*". Once the deployment gets stuck, I only find the below ressources in my subscription. 

<img width="679" alt="Screenshot 2022-12-22 at 15 12 06" src="https://user-images.githubusercontent.com/243504/209152651-a539f086-c4ed-41a8-873b-d2291d2d1a5b.png">

I tried to deploy to `switzerlandnorth`. I will now go ahead and try westeurope.

**Comment by SomeBdyElse** [comment]
`westeurope` did not do the trick either. I will check with the Azure support whether there are any issues with our subscriptions. Unfortunately I do not get any error or log messages (except deployment timeout), so I do not have anything to work on. 

@sebafo as the issue seems unrelated to the code provided in this repo, I think that we can close the issue. If I get any insights that lead to code changes, I will let you know.

**Comment by sebafo** [comment]
Perhaps you can try one thing:
Deploy Container Apps via the Azure Portal and check whether it works that way. Because there is really not that much magic in the Container Apps Environment deployment in this repo.

Yes, let me know if you have any insights that lead to code changes.

For now, I close the issue as requested.

**Comment by SomeBdyElse** [comment]
JFYI I ran the deployment on a clean Azure subscription and it succeeded. I had to add some resource providers (https://github.com/sebafo/frontdoor-container-apps/issues/3), but that was no big deal. 

Now I see 3 resource groups. 
* news-hub-rg <-- the subject of this deployment
* NetworkWatcherRG
* MC_…-rg_…_westeurope

I imagine that the creation of the two additional ressource groups was prohibited by an organizational policy that requires resource groups to have certain tags. I will try to get rid of those policies or create the additional resource groups manually. This issue can remain closed.
